BlazarService/src/main/java/com/hubspot/blazar/util/SlackClientWrapper.java

```diff
@@ -0,0 +1,129 @@
+package com.hubspot.blazar.util;
+
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.rholder.retry.Retryer;
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import com.google.common.base.Optional;
+import com.google.common.base.Predicates;
+import com.google.inject.Inject;
+import com.ullink.slack.simpleslackapi.SlackAttachment;
+import com.ullink.slack.simpleslackapi.SlackChannel;
+import com.ullink.slack.simpleslackapi.SlackMessageHandle;
+import com.ullink.slack.simpleslackapi.SlackSession;
+import com.ullink.slack.simpleslackapi.SlackUser;
+import com.ullink.slack.simpleslackapi.replies.ParsedSlackReply;
+import com.ullink.slack.simpleslackapi.replies.SlackMessageReply;
+
+/**
+ * This class handles all the details of sending a slack message, and verifying that it did send.
+ */
+public class SlackClientWrapper {
```

Comment:
Doesn't have to be since it doesn't do anything singleton-y since its using the single `SlackSession` instance. From a clarity point of view would make sense to make this a singleton.